File: app/controllers/Application.java
package controllers;

import play.api.mvc.Action;
import play.api.mvc.AnyContent;
import play.mvc.Controller;
import play.mvc.Result;
import views.html.index;

public class Application extends Controller {

    public static Result index() {
        return ok(index.render());
    }

    public static Action<AnyContent> webJarAssetsAtLocation(final String file) {
        return controllers.WebJarAssets.at(controllers.WebJarAssets
                .locate(file));
    }
}

File: app/controllers/Application.scala
package controllers

import play.api.mvc.{Action, Controller}
import play.api.http.MimeTypes

object Application extends Controller {

  def dynamicIndex = Action {
    Ok(views.html.DynamicIndex())
  }
  
  // returns the JavaScript used to setup RequireJS for WebJars
  def setupRequireJs = Action {
    Ok(org.webjars.RequireJS.getSetupJavaScript(routes.WebJarAssets.at("").url)).as(MimeTypes.JAVASCRIPT)
  }
  
}
